backend/src/middleware/challenge.ts

```diff
@@ -0,0 +1,50 @@
+import { NextFunction, Request, Response } from 'express';
+import { sendLoginMail } from '../controllers/mailer.js';
+import { userModel } from '../models/user.schema.js';
+import { compareSync } from 'bcrypt';
+
+export const store : Map<string, number> = new Map();
+
+export function challengeMiddleware(request : Request, response : Response, next : NextFunction) : void
+{
+	const { email, challenge, password } = request.body;
+
+	userModel.findOne(
+		{
+			email
+		})
+		.then(user =>
+		{
+			return { compare: compareSync(password, user.password), user };
+		}).then(result =>
+		{
+			if (result.compare)
+			{
+				if (!challenge)
+				{
+					const createdChallenge : number = createChallenge();
+
+					store.set(email, createdChallenge);
+					sendLoginMail(email, createdChallenge);
+
+					response.status(200).json(
+					{
+						action: 'request-challenge'
+					});
+				}
+				else
+				{
+					next();
+				}
+			}
+			else
+			{
+				response.status(401).json((error : Error) =>error.message);
```

Comment:
Please search the whole backend for pattern like that... there is no error here! you just cast the body object to a wrong type and then return nothing...

Dont mix it with catch(error => ...